File: MezonQuizFE/src/Interface/category.dto.ts
export interface CategoryDto extends SaveCategoryDto {
  id: string;
  createdAt: string;
}

export interface SaveCategoryDto {
    name: string;
    slug: string;
    icon?: string;
    sortOrder?: number;
}

export type createCategoryResponse = CategoryDto;
export type updateCategoryResponse = CategoryDto;
export type deleteCategoryResponse = void;

File: MezonQuiz/src/WebApp/Application/Categories/Dtos/CategoryDto.cs
using System.ComponentModel.DataAnnotations;

namespace WebApp.Application.Categories.Dtos
{
    public class CategoryDto : SaveCategoryDto
    {
        public Guid Id { get; set; }
        public DateTime CreatedAt { get; set; }
    }

    public class SaveCategoryDto
    {
        public string Name { get; set; } = null!;
        public string? Slug { get; set; }
        public string? Icon { get; set; }
        public int SortOrder { get; set; }

        public void Validate()
        {
            if (string.IsNullOrWhiteSpace(Name))
            {
                throw new ArgumentException("Category name is required.", nameof(Name));
            }
            else if (SortOrder < 0)
            {
                throw new ArgumentException("Category sort order must be non-negative.", nameof(SortOrder));
            }
        }

    }
}
